plugins/Observability/src/Observability.cpp

```diff
@@ -208,23 +207,33 @@ void Observability::init()
 	addAction("actionShow_Observability",        N_("Observability"), N_("Observability"),      "flagShowReport");
 	addAction("actionShow_Observability_dialog", N_("Observability"), N_("Show settings dialog"), configDialog, "visible", ""); // Allow assign shortkey
 
-	StelGui* gui = dynamic_cast<StelGui*>(StelApp::getInstance().getGui());
-	if (gui!=Q_NULLPTR)
-	{
-		button = new StelButton(Q_NULLPTR,
-					QPixmap(":/observability/bt_observab_on.png"),
-					QPixmap(":/observability/bt_observab_off.png"),
-					QPixmap(":/graphicGui/miscGlow32x32.png"),
-					"actionShow_Observability",
-					false,
-					"actionShow_Observability_dialog");
-		gui->getButtonBar()->addButton(button, "065-pluginsGroup");
-	}
-	
-	updateMessageText();
-	connect(&StelApp::getInstance(), SIGNAL(languageChanged()),
-	        this, SLOT(updateMessageText()));
-	connect(StelApp::getInstance().getCore(), SIGNAL(configurationDataSaved()), this, SLOT(saveConfiguration()));
+   StelGui * gui = dynamic_cast<StelGui *>(StelApp::getInstance().getGui());
+   if (gui != Q_NULLPTR) {
+      button = new StelButton(Q_NULLPTR,
+			      QPixmap(":/observability/bt_observab_on.png"),
+			      QPixmap(":/observability/bt_observab_off.png"),
+			      QPixmap(":/graphicGui/miscGlow32x32.png"),
+			      "actionShow_Observability",
+			      false,
+			      "actionShow_Observability_dialog");
+      gui->getButtonBar()->addButton(button, "065-pluginsGroup");
+   }
+
+   updateMessageText();
+   connect(&StelApp::getInstance(), &StelApp::languageChanged, this, &Observability::onLanguageChanged);
+   connect(StelApp::getInstance().getCore(), SIGNAL(configurationDataSaved()), this, SLOT(saveConfiguration()));
+
+   connect(this, &Observability::flagReportVisibilityChanged, this, [&](bool enabled) {
+      if (enabled) {
+        qDebug() << "[Observability] Creating plugin status dependent connections.";
```

Comment:
You can comment these two debug messages away in non-debug builds. Either #ifndef NDEBUG or just comment away when they are working.